src/request.cc

```diff
@@ -162,6 +162,7 @@ BaseUrl::BaseUrl(std::string host, bool https, std::string region)
     return;
   }
 
+  this->https = url.https;
```

Comment:
This is not correct. `this->https` is set in the constructor definition already.

Reply:
I have seen that, but the Url class parses the schema to set the https flag in the url class that…
Correct me if I am wrong, but if I am constructing a u=BasicUrl("https://foo.bar", false)
Then u.https is false

Reply:
Firstly `BaseUrl` accepts `host`. If URL is used, it takes only host and port. Secondly passed argument should override values. The current behavior is correct and this change is not correct.

Reply:
I don't think this is a normal operation, in fact, I happened to experience this problem and have been troubled for a while for using of an HTTP address but the constructed baseURL being the default HTTPS. I cannot understand choosing to pass parameters instead of using URL parsed values. After all, people are sensitive to their own URLs and are not aware of the construction of functions in third-party libraries

Reply:
The right fix is to fail if URL is passed like `minio-go` and `minio-py`. We should not support URL style as base URL